### PR TITLE
fix: handle empty partOf

### PR DESCRIPTION
**Pull request by Zangetsu101**
~The location resource created corresponding to a person's address doesn't have the `partOf` entry~ International addresses don't have a `partOf` field for their corresponding location resource. So we needed to handle them separately when transforming the `partOf.reference` field as `{ partOf: {} }` was getting added to all those location resources in the new view

### CODE REVIEW

File: packages/migration/src/migrations/hearth/20240508142533-location-view-with-plain-ids.ts [hunk 1]
```diff
@@ -17,22 +17,30 @@ export const up = async (db: Db, client: MongoClient) => {
     pipeline: [
       {
```
Comment (rikukissa) [inline packages/migration/src/migrations/hearth/20240508142533-location-view-with-plain-ids.ts:18]: @Zangetsu101 as this is slightly related to
https://github.com/opencrvs/opencrvs-core/issues/7180

I wonder if we could completely filter out the home addresses from this view if those are the ones without partOf properly set up?

Comment (Zangetsu101) [reply packages/migration/src/migrations/hearth/20240508142533-location-view-with-plain-ids.ts:18]: yea makes sense. but then the bundle won't resolve those location resources :grimacing: You know `Encounter.location.location.reference `

Comment (rikukissa) [reply packages/migration/src/migrations/hearth/20240508142533-location-view-with-plain-ids.ts:18]: Good point. Ok, let's proceed with this for now 👍

**Review by rikukissa — APPROVED** [review]

File: packages/migration/src/migrations/hearth/20240508142533-location-view-with-plain-ids.ts (hunk 1 above)
Comment (Zangetsu101) [reply packages/migration/src/migrations/hearth/20240508142533-location-view-with-plain-ids.ts:18]: you know what, we probably don't need to worry about that. let me try it out

Comment (Zangetsu101) [reply packages/migration/src/migrations/hearth/20240508142533-location-view-with-plain-ids.ts:18]: There is potentially some other bug here. :sweat_smile: The `partOf` field is not there when `{ 'type.coding.code': "OTHER" }` but it's there for the value "PRIVATE_HOME"

Comment (Zangetsu101) [reply packages/migration/src/migrations/hearth/20240508142533-location-view-with-plain-ids.ts:18]: Okay I get it now, it's not there for international addresses which makes sense. So let's keep it as is now

**Pull request merged**